sqlglot/generator.py

```diff
@@ -2806,6 +2809,12 @@ def windowspec_sql(self, expression: exp.WindowSpec) -> str:
             csv(self.sql(expression, "end"), self.sql(expression, "end_side"), sep=" ")
             or "CURRENT ROW"
         )
+
+        if self.SUPPORTS_WINDOW_EXCLUDE:
+            exclude = self.sql(expression, "exclude")
+            if exclude:
+                return f"{kind} BETWEEN {start} AND {end} EXCLUDE {exclude}"
+
         return f"{kind} BETWEEN {start} AND {end}"
```

Comment:
This is how we usually implement generation logic:

```suggestion
        window_spec = f"{kind} BETWEEN {start} AND {end}"

        exclude = self.sql(expression, "exclude")
        if exclude:
            if self.SUPPORTS_WINDOW_EXCLUDE:
                window_spec += f" EXCLUDE {exclude}"
            else:
                self.unsupported("EXCLUDE clause is not supported in the WINDOW clause")

        return window_spec
```